crates/traits/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::dispatch::DispatchError;
+use num_bigint::{BigUint, ToBigUint};
+
+use primitives::{CurrencyId, PriceDetail};
+
+pub mod loans;
+pub use loans::*;
+
+pub trait PriceFeeder {
+    fn get_price(asset_id: &CurrencyId) -> Option<PriceDetail>;
+}
+
+pub trait ConvertToBigUint {
+    fn get_big_uint(&self) -> BigUint;
+}
+
+impl ConvertToBigUint for u128 {
+    fn get_big_uint(&self) -> BigUint {
+        self.to_biguint().unwrap()
```

Comment:
Is this safe to do? When can `to_biguint` fail?

Reply:
This snippet is unchanged from the original code. It can fail if the `u128` doesn't fit inside a `u64` during the conversion:
- Step 1: https://docs.rs/num-bigint/0.4.0/src/num_bigint/biguint/convert.rs.html#571
- Step 2: https://docs.rs/num-bigint/0.4.0/src/num_bigint/biguint/convert.rs.html#548
- Step 3: https://docs.rs/num-traits/0.2.14/src/num_traits/cast.rs.html#498

Reply:
When let's make it return Result, as we do for other type conversions?